data_diff/diff_tables.py

```diff
@@ -346,12 +348,16 @@ def _validate_and_adjust_columns(self, table1, table2):
                 table1._schema[c] = col1.replace(precision=lowest.precision)
                 table2._schema[c] = col2.replace(precision=lowest.precision)
 
+            elif isinstance(col1, StringType):
+                if not isinstance(col2, StringType):
+                    raise TypeError(f"Incompatible types for column '{c}':  {col1} <-> {col2}")
```

Comment:
Will repeat my pet-peeve of always including database names, because I've seen this error and scratched my head so many times 😅 I know it's a bit less urgent here, but a good rule for us to follow I think. Maybe something like `Postgres/varchar(100)` would probably be fine.

Reply:
Yeah, I agree. In general I should do another pass on error messages.